server/coordinator/scheduler/scheduler_manager.go

```diff
@@ -110,6 +113,19 @@ func (m *ManagerImpl) Start(ctx context.Context) error {
 				// Get latest cluster snapshot.
 				clusterSnapshot := m.clusterMetadata.GetClusterSnapshot()
 				log.Debug("scheduler manager invoke", zap.String("clusterSnapshot", fmt.Sprintf("%v", clusterSnapshot)))
+
+				// TODO: Perhaps these codes related to deployMode need to be refactored.
+				// If deployMode is turned on, the scheduler will only be scheduled in the non-stable state.
+				if m.deployMode && clusterSnapshot.Topology.ClusterView.State == storage.ClusterStateStable {
+					continue
+				}
+				if clusterSnapshot.Topology.ClusterView.State == storage.ClusterStatePrepare && len(clusterSnapshot.Topology.ShardViewsMapping) == len(clusterSnapshot.Topology.ClusterView.ShardNodes) {
+					if err := m.clusterMetadata.UpdateClusterView(ctx, storage.ClusterStateStable, clusterSnapshot.Topology.ClusterView.ShardNodes); err != nil {
```

Comment:
It's weird to do cluster state update here? Is there any better place to do this?

Reply:
I also think the implementation here is a bit dirty, but for now, it can only be placed here. This corresponds to the todo I left above, and it will be modified when the code related to deployMode is refactored later.